Grammar/python.gram

```diff
@@ -331,14 +333,19 @@ kwds[arg_ty]: '**' a=param_no_default { a }
 param_no_default[arg_ty]:
     | a=param ',' tc=TYPE_COMMENT? { _PyPegen_add_type_comment_to_arg(p, a, tc) }
     | a=param tc=TYPE_COMMENT? &')' { _PyPegen_add_type_comment_to_arg(p, a, tc) }
+param_no_default_star_annotation[arg_ty]:
```

Comment:
Why not join these to `param_no_default`? That way we can remove the extra rule in `star_etc`.